### PR TITLE
Add basic test infrastructure 

### PR DESCRIPTION
**Pull request by awisniew90**
Add test infrastructure to stack repo using github actions. 

More details here: https://github.com/awisniew90/application-stack/blob/regression_test/test/README.md

### CODE REVIEW

File: templates/outer-loop/Dockerfile [hunk 1]
```diff
@@ -33,10 +33,14 @@ FROM {{.OL_UBI_IMAGE}}
 # 2a) Copy user defined shared resources 
 COPY --from=compile --chown=1001:0 /work/shared /opt/ol/wlp/usr/shared/
 
+RUN true
```
Comment (scottkurz) [inline templates/outer-loop/Dockerfile:36]: This `RUN true` looks hackish... if we're essentially doing this just for the GH Actions env, could we put a placeholder comment here, and then find/replace before we run the docker build in the test env?   (Like we do  a sed substitution in the deploy yaml).

File: test/README.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+Applying URL changes
+ ✗  Failed To Update Config To Component Deployed.
+Error: unable to create ingress: error creating ingress: Internal error occurred: failed calling webhook "validate.nginx.ingress.kubernetes.io": Post https://ingress-nginx-controller-admission.kube-system.svc:443/extensions/v1beta1/ingresses?timeout=30s: dial tcp 10.96.251.20:443: connect: connection refused
+```
+
+1. Warning message due to known "bug" in `softprops/diffset@v1` plugin (https://github.com/softprops/diffset/issues/5):
+
+```
+Unexpected input(s) 'stackimage_files', valid inputs are ['base']
+```
```
Comment (scottkurz) [inline test/README.md:68]: Sounds like this is just explaining some noise but the test is still valid, and passes... i.e. doesn't need to be rerun?

File: test/README.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+## Known issues
+
+1. Intermittent Ingress connection refused errors:
+
+```
+Applying URL changes
+ ✗  Failed To Update Config To Component Deployed.
+Error: unable to create ingress: error creating ingress: Internal error occurred: failed calling webhook "validate.nginx.ingress.kubernetes.io": Post https://ingress-nginx-controller-admission.kube-system.svc:443/extensions/v1beta1/ingresses?timeout=30s: dial tcp 10.96.251.20:443: connect: connection refused
+```
```
Comment (scottkurz) [inline test/README.md:62]: What's the PR author supposed to do (take action?) if this happens?  Click and re-run?   Does it generally affect all tests or a subset ?

File: test/README.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
+## Test Overview
```
Comment (scottkurz) [inline test/README.md:1]: Wow, nice write-up .. appreciate this !  We could move to the Wiki or even a gh_pages branch if we end up changing frequently.

File: test/stack-test-inner-loop.sh [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+echo -e "\n> Create URL with Minikube IP"
+odo url create --host $(minikube ip).nip.io
+
+echo -e "\n> Push to Minikube"
+odo push
+
+echo -e "\n> Check for server start"
+count=1
+while ! odo log | grep -q "The defaultServer server is ready to run a smarter planet"; do
```
Comment (scottkurz) [inline test/stack-test-inner-loop.sh:27]: OK, though better to grep for CWWKF0011I  id

File: .github/workflows/stack-regression-tests.yml [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        id: outer-loop-diff
+        uses: softprops/diffset@v1
+        with:
+          base: main
+          outerloop_files: |
+            stackimage/*
+            templates/stackimage/*
+            templates/outer-loop/*
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (scottkurz) [inline .github/workflows/stack-regression-tests.yml:35]: Just for my own education, I'm curious:  what does this do?

File: test/stack-test-outer-loop.sh [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+echo -e "\n> Copy Dockerfile"
+cp ../../generated/Dockerfile Dockerfile
+
+echo -e "\n> Copy app-deploy.yaml"
+cp ../../templates/outer-loop/app-deploy.yaml app-deploy.yaml
+
+echo -e "\n> Create new odo project"
+odo project create ${OUTER_LOOP_TEST_NAMESPACE}
+
+### This is only needed if the Minikube driver is "docker". As of now, we are using diver=none or "bare metal"
```
Comment (scottkurz) [inline test/stack-test-outer-loop.sh:21]: Nice.. thx for explaining since this is something I was unsure of.

**Review by scottkurz — CHANGES_REQUESTED** [review]
Nice... most of these are just questions.  Esp. appreciate the doc.  My general thought is to just merge in and make any other adjustments later.   The only reason I want to change first is the `RUN true` in the Dockerfile looks a bit too hacked if we can change it quick.

File: test/README.md (hunk 2 above)
Comment (awisniew90) [reply test/README.md:68]: right - these are just benign warnings. tests wont have to be rerun.

File: test/README.md (hunk 3 above)
Comment (awisniew90) [reply test/README.md:62]: Yea the tests need to be rerun (I will add that in).  If it starts to happen often, we can open an issue and look into it, but I've only seen it a handful of times. perhaps a timing problem.

File: .github/workflows/stack-regression-tests.yml (hunk 6 above)
Comment (awisniew90) [reply .github/workflows/stack-regression-tests.yml:35]: I'm honestly not sure... It's a "required" prop for the plugin, but it doesn't elaborate on it more that "GITHUB_TOKEN as provided by secrets"

**Review by scottkurz — APPROVED** [review]
LGTM.  Thanks !

**Pull request merged**